cmd/query/app/server.go

```diff
@@ -79,14 +81,35 @@ func createHTTPServer(querySvc *querysvc.QueryService, queryOpts *QueryOptions,
 
 	apiHandler.RegisterRoutes(r)
 	RegisterStaticHandler(r, logger, queryOpts)
-	compressHandler := handlers.CompressHandler(r)
+	var compressHandler http.Handler
+	if queryOpts.BearerTokenPropagation {
+		compressHandler = handlers.CompressHandler(BearTokenPropagationHandler(logger, r))
+	} else {
+		compressHandler = handlers.CompressHandler(r)
```

Comment:
avoid duplicating wrappers. There's no business meaning of calling CompressHandler twice in different context.

I suggest `var h http.Hander = r` and then keep overwriting `h` with every new decorator (including in L90)